pom.xml

```diff
@@ -255,6 +245,34 @@
   <build>
     <pluginManagement>
       <plugins>
+        <plugin>
+          <groupId>com.diffplug.spotless</groupId>
+          <artifactId>spotless-maven-plugin</artifactId>
+          <configuration>
```

Comment:
you do not need to rewrite all the configuration but only inherits from parent and add elements.
would be better to simply use:
```
          <configuration>
            <java combine.children="append">
              <toggleOffOn>
                <off>@formatter:off</off>
                <on>@formatter:on</on>
              </toggleOffOn>
            </java>
            <pom combine.children="append">
              <toggleOffOn>
                <off>@formatter:off</off>
                <on>@formatter:on</on>
              </toggleOffOn>
            </pom>
          </configuration>
```

Reply:
I know this (`combine.children`) works for collection, but also for configuration elements bind by POJO ?

Reply:
`mvn help:effective-pom` with `combine.children="append"` is telling me that it does not work :sob:

Reply:
it's just xml elements to add or override. The type of what it's mapped to doesn't matter. The mapping is happening later. at the stage of XML merging/inheritance Maven doesn'r have any idea about the target class.

just use ` mvn help:effective-pom` to see the result.

Reply:
ok, got it :partying_face: 

`pluginManagement`  vs `plugins`

Reply:
```
olamy@pop-os:~/dev/sources/jenkins/pipeline-maven-plugin$ git diff
diff --git a/pom.xml b/pom.xml
index ca0e948..46383f5 100644
--- a/pom.xml
+++ b/pom.xml
@@ -249,23 +249,13 @@
           <groupId>com.diffplug.spotless</groupId>
           <artifactId>spotless-maven-plugin</artifactId>
           <configuration>
-            <java>
+            <java combine.children="append">
               <toggleOffOn>
                 <off>@formatter:off</off>
                 <on>@formatter:on</on>
               </toggleOffOn>
             </java>
-            <pom>
-              <sortPom>
-                <nrOfIndentSpace>2</nrOfIndentSpace>
-                <predefinedSortOrder>recommended_2008_06</predefinedSortOrder>
-                <sortProperties>true</sortProperties>
-                <sortModules>true</sortModules>
-                <sortExecutions>true</sortExecutions>
-                <sortDependencies>groupId,artifactId</sortDependencies>
-                <sortDependencyExclusions>groupId,artifactId</sortDependencyExclusions>
-                <sortPlugins>groupId,artifactId</sortPlugins>
-              </sortPom>
+            <pom combine.children="append">
               <toggleOffOn>
                 <off>@formatter:off</off>
                 <on>@formatter:on</on>

```
then
```
mvn help:effective-pom > eff.pom
```
see output here: https://gist.github.com/olamy/fb163f2380f8a1475365838a73de2cb7#file-eff-pom-L2630
as far as I can see it works.